test/unittest/dds/xtypes/dyndata_json/types/comprehensive_type/ComprehensiveTypeFilled_OMG.hpp

```diff
@@ -0,0 +1,335 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+std::string expected_json_comprehensive_filled_omg = R"({
+    "complex_array": [
+        {
+            "bitmask_sequence": [],
+            "complex_union": {
+                "fourth": {
+                    "second": 0
+                }
+            },
+            "inner_union": {
+                "second": 0
+            },
+            "long_array": [
+                [
+                    [
+                        0,
+                        0,
+                        0,
+                        0
```

Comment:
I am afraid we should be filling the whole complex structure, to cover cases like a sequence of for example unions or other complex objects that otherwise would not be printed.